site/src/pages/WorkspacePage/WorkspaceReadyPage.tsx

```diff
@@ -33,88 +28,74 @@ import { templateVersion, templateVersions } from "api/queries/templates";
   builds: TypesGen.WorkspaceBuild[] | undefined;
   buildsError: unknown;
   onLoadMoreBuilds: () => void;
   isLoadingMoreBuilds: boolean;
   hasMoreBuilds: boolean;
-  canAutostart: boolean;
 }
 
 export const WorkspaceReadyPage = ({
-  workspaceState,
-  workspaceSend,
-  quota,
+  workspace,
+  template,
+  permissions,
   builds,
   buildsError,
   onLoadMoreBuilds,
   isLoadingMoreBuilds,
   hasMoreBuilds,
-  canAutostart,
 }: WorkspaceReadyPageProps): JSX.Element => {
+  const navigate = useNavigate();
+  const queryClient = useQueryClient();
   const { buildInfo } = useDashboard();
   const featureVisibility = useFeatureVisibility();
-  const {
-    workspace,
-    template,
-    templateVersion: currentVersion,
-    deploymentValues,
-    buildError,
-    cancellationError,
-    sshPrefix,
-    permissions,
-    missedParameters,
-  } = workspaceState.context;
   if (workspace === undefined) {
     throw Error("Workspace is undefined");
   }
-  const deadline = getDeadline(workspace);
-  const canUpdateWorkspace = Boolean(permissions?.updateWorkspace);
-  const canUpdateTemplate = Boolean(permissions?.updateTemplate);
-  const canRetryDebugMode =
-    Boolean(permissions?.viewDeploymentValues) &&
-    Boolean(deploymentValues?.enable_terraform_debug_mode);
-  const favicon = getFaviconByStatus(workspace.latest_build);
-  const navigate = useNavigate();
-  const [changeVersionDialogOpen, setChangeVersionDialogOpen] = useState(false);
-  const [isConfirmingUpdate, setIsConfirmingUpdate] = useState(false);
-  const [confirmingRestart, setConfirmingRestart] = useState<{
-    open: boolean;
-    buildParameters?: TypesGen.WorkspaceBuildParameter[];
-  }>({ open: false });
 
-  const { data: allVersions } = useQuery({
-    ...templateVersions(workspace.template_id),
-    enabled: changeVersionDialogOpen,
-  });
-  const { data: latestVersion } = useQuery({
-    ...templateVersion(workspace.template_active_version_id),
-    enabled: workspace.outdated,
+  // Debug mode
+  const { data: deploymentValues } = useQuery({
+    ...deploymentConfig(),
+    enabled: permissions?.viewDeploymentValues,
   });
-  const [faviconTheme, setFaviconTheme] = useState<"light" | "dark">("dark");
-  useEffect(() => {
-    if (typeof window === "undefined" || !window.matchMedia) {
-      return;
-    }
+  const canRetryDebugMode = Boolean(
+    deploymentValues?.config.enable_terraform_debug_mode,
+  );
 
-    const isDark = window.matchMedia("(prefers-color-scheme: dark)");
-    // We want the favicon the opposite of the theme.
-    setFaviconTheme(isDark.matches ? "light" : "dark");
-  }, []);
+  // Build logs
   const buildLogs = useWorkspaceBuildLogs(workspace.latest_build.id);
   const shouldDisplayBuildLogs =
     hasJobError(workspace) ||
     ["canceling", "deleting", "pending", "starting", "stopping"].includes(
       workspace.latest_build.status,
     );
+
+  // Restart
+  const [confirmingRestart, setConfirmingRestart] = useState<{
```

Comment:
Wondering if the state type could be defined as a discriminated union, so that you're guaranteed to have `buildParameters` be defined whenever `open` is true

Though I guess that would also require updating the type definitions further down the tree

Reply:
I usually think discriminated unions are always a good idea but for this case, I would just let it as it is since there is not too much value and to avoid more changes.